react/week4/git-api/src/components/UserContext.js

```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useEffect, useState } from "react"
+
+const API = `https://api.github.com/search/users?q=`
+
+export const UserContext = createContext()
+
+export const UserProvider = ({ children, input, loading, setLoading }) => {
```

Comment:
It's great that you noticed that you can extract this logic into a separate component. Now the App component is much more transparent. Good thinking!